src/ppom-controller.ts

```diff
@@ -345,129 +361,216 @@ export class PPOMController extends BaseControllerV2<
    * @returns True if PPOM data requires update.
    */
   async #shouldUpdate(): Promise<boolean> {
-    if (this.state.newChainId !== this.state.lastChainId) {
-      return true;
+    const { chainId, chainIdCache } = this.state;
+
+    if (
+      chainIdCache.find(
+        ({ chainId: cid, dataFetched }) => cid === chainId && dataFetched,
+      )
+    ) {
+      return false;
     }
 
-    return this.#ppom === undefined;
+    return true;
```

Comment:
If we only return a boolean either way, could we just cast the return value from `find`?

Reply:
yeh this was just to make it more descriptive, I updated the PR